### DISCUSSION TITLE
Regex Manager + Artifactory with package files named <name>-<version>.tar.gz

### DISCUSSION THREAD

**Discussion by lcts** [renovatebot/renovate#21473] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.41.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab CE 15.9.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm trying to use Renovate to update some packages that are fetched from an Artifactory instance.
The packages are stored in Artifactory in the following format:
```
<artifactory url>/<artifactory repo>/download/<package>-<version>.tar.gz
```
and I'm trying to create a Regex Manager that can renovate these types of dependencies.

### What you tried so far.

I've already written a couple of Regex Managers for urls that have package name and version in seperate folders, i.e. `.../<package>/<something containing the version>` which works really nicely in conjunction with Renovate's templating for variables. However the `<fixed folder>/<file with package and version>`-Format seems to cause trouble:

I've used the following:
```
"regexManagers": [
    {
      "fileMatch": "<some file>",
      "matchStrings": [
        "https:\\/\\/my\\.artifactory\\.url(:443)?\\/artifactory\\/my-repo\\/download\\/(?<artifactName>.*-.*?)-(?<currentValue>.*?)\\.tar.gz"
      ],
      "packageNameTemplate": "download",
      "extractVersionTemplate": "{{{artifactName}}}-(?<version>.*?)\\.tar.gz",
      "depNameTemplate": "{{{artifactName}}}",
      "registryUrl": "https://my.artifactory.url/artifactory/my-repo",
      "datasourceTemplate": "artifactory",
      "versioningTemplate": "loose"
    }
  ],
```
in the hope of making the Artifactory datasource look the list of files in the `.../download` folder and then extract the actual versions of the actual package with the `extractVersionTemplate`.
Renovate appears to do all the extractions correctly, however if fails to look up the dependency in Artifactory:

```
# Output of renovate run with LOGLEVEL=DEBUG
{
                 "depName": "my-package",
                 "packageName": "download",
                 "currentValue": "6.1.0",
                 "datasource": "artifactory",
                 "versioning": "loose",
                 "extractVersion": "my-package-(?<version>.*?)\\.tar.gz",
                 "replaceString": "https://my.artifactory.url/artifactory/my-repo/download/my-package-6.1.0.tar.gz",
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "download",
                     "message": "Failed to look up artifactory package download"
                   }
                 ]
               }
```

I looked at the code for the Artifactory datasource, it seems like it just queries `{{ registryURL}}/{{ packageName}}` to which Artifactory replies with the list of files contained in that path. In my case this would be something like
```
my-package-5.1.0.tar.gz
my-package-6.1.0.tar.gz
my-package-7.1.0.tar.gz
my-other-package-4.3.2.tar.gz
my-other-package-7.0.0.tar.gz
...
```
so I don't see why `"extractVersion": "mypackage-(?<version>.*?)\\.tar.gz"` doesn't seem to work or why it claims to not be able to look up the package 'download' when curling the url works perfectly fine.

*So, any ideas as to what is going wrong here?*

And, for my understanding, is it correct that
- depName is what Renovate uses to track packages/label MRs etc
- currentValue is what Renovate uses as the currently used version
- packageName is what is looked up in the datasource, it defaults to depName if not given
- extractVersion is how renovate extracts a version from the answer provided by the datasource, with version being the actual vaule to which currentValue is compared to
- the artifactory datasource (by default without any templating etc.) queries registryURL/packageName and treats the list of entries in the replies as versions?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by lcts** [comment]
The error message is thrown by [getPackageReleases](https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/index.ts#L343), I guess because [this bit](https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/index.ts#L355-L372) doesn't return any results, but from there I don't know enough about how Renovate works to further track down what goes wrong.
My code example above still looks 100% fine to me.

**Reply by rarkins** [reply]
If you're unable to create a public reproduction, the best thing would be for you to debug this locally to see at which point the releases are filtered out

**Reply by lcts** [reply]
Hm, I can post the full debug log - nothing particularly secret about this - but I don't think I'll be able to start hosting a publicly accessible Artifactory instance to test this, no.

Before I go off on a wild-goose chase, am I correct about this:

> depName is what Renovate uses to track packages/label MRs etc
> currentValue is what Renovate uses as the currently used version
> packageName is what is looked up in the datasource, it defaults to depName if not given
> extractVersion is how renovate extracts a version from the answer provided by the datasource, with version being the actual value to which currentValue is compared to
> the artifactory datasource (by default without any templating etc.) queries registryURL/packageName and treats the list of entries in the replies as versions

**Reply by rarkins** [reply]
depName is the "pretty" name. Often the same as packageName.

currentValue is the literal string which defines a version or version range/constraint.

extractVersion is optional and rarely used, only really needed if the registry returns non-versions like in your case.

Also by "debugging" I don't mean debug logs, I mean debugging (e.g. running in VSCode or Chrome)
